identity/src/main/java/com/stripe/android/identity/ui/IdentityTheme.kt

```diff
@@ -25,48 +26,77 @@ import com.stripe.android.uicore.StripeThemeDefaults
 import com.stripe.android.uicore.StripeTypography
 import com.stripe.android.uicore.toComposeTypography
 import java.lang.reflect.Method
+import com.google.accompanist.themeadapter.material.R as MaterialR
+import com.google.accompanist.themeadapter.material3.R as Material3R
 
 /**
  * IdentityTheme tries to read theme values from hosting app's context. Then adapt the values with
  * CompositionLocalProviders required by StripeTheme.
  */
 @Composable
 internal fun IdentityTheme(content: @Composable () -> Unit) {
-    val context = LocalContext.current
-    val key = context.theme.key ?: context.theme
-
-    val layoutDirection = LocalLayoutDirection.current
-
-    val themeParams = remember(key) {
-        createMdcTheme(
-            context = context,
-            layoutDirection = layoutDirection,
-            readColors = true,
-            readTypography = true,
-            readShapes = true,
-            setTextColors = false,
-            setDefaultFontFamily = false
-        )
-    }
+    val themeParameters = extractThemeParameters()
 
     val isRobolectricTest = runCatching {
         Build.FINGERPRINT.lowercase() == "robolectric"
     }.getOrDefault(false)
 
     val inspectionMode = LocalInspectionMode.current || isRobolectricTest
-    val hostingAppColors = themeParams.colors ?: MaterialTheme.colors
-    val hostingAppTypography = themeParams.typography ?: MaterialTheme.typography
-    val hostingAppShapes = themeParams.shapes ?: MaterialTheme.shapes
 
     AdoptForStripeTheme(
-        hostingAppColors = hostingAppColors,
-        hostingAppTypography = hostingAppTypography,
-        hostingAppShapes = hostingAppShapes,
+        hostingAppColors = themeParameters.colors,
+        hostingAppTypography = themeParameters.typography,
+        hostingAppShapes = themeParameters.shapes,
         inspectionMode = inspectionMode,
         content = content
     )
 }
 
+/**
+ * Extracts theme parameters from the hosting app's context
+ */
+@Composable
+private fun extractThemeParameters(): ThemeParameters {
+    val context = LocalContext.current
+    val layoutDirection = LocalLayoutDirection.current
+    val key = context.theme.key ?: context.theme
+
+    val defaultParameters = ThemeParameters(
+        colors = MaterialTheme.colors,
+        typography = MaterialTheme.typography,
+        shapes = MaterialTheme.shapes
+    )
```

Comment:
Use default Material theme if theme params cannot be extracted from a `MaterialComponents` theme.